service/rtc/config.go

```diff
@@ -55,3 +61,99 @@ func (c SessionConfig) IsValid() error {
 
 	return nil
 }
+
+type ICEServerConfig struct {
+	URLs       []string `toml:"urls" json:"urls"`
+	Username   string   `toml:"username,omitempty" json:"username,omitempty"`
+	Credential string   `toml:"credential,omitempty" json:"credential,omitempty"`
+}
+
+type ICEServers []ICEServerConfig
+
+func (c ICEServerConfig) IsValid() error {
+	if len(c.URLs) == 0 {
+		return fmt.Errorf("invalid empty URLs")
+	}
+	for _, u := range c.URLs {
+		if u == "" {
+			return fmt.Errorf("invalid empty URL")
+		}
+
+		if !strings.HasPrefix(u, "stun:") && !strings.HasPrefix(u, "turn:") {
+			return fmt.Errorf("URL is not a valid STUN/TURN server")
+		}
+	}
+	return nil
+}
+
+func (s ICEServers) IsValid() error {
+	for _, cfg := range s {
+		if err := cfg.IsValid(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s ICEServers) getSTUN() string {
+	for _, cfg := range s {
+		for _, u := range cfg.URLs {
+			if strings.HasPrefix(u, "stun:") {
+				return u
```

Comment:
I guess that's expected, to return only the first?

Reply:
Yes, we just need one. We could randomize but not a big deal at the moment.

> Cool. I'm assuming there's no breaking changing to client config, so we don't have to worry about that. :)

On the client, I am still finishing up the PRs but we'll be adding a new ICEServerConfigs field along with the existing ICEServers which we'll deprecate. Will have PRs up tomorrow.